core/services/feeds/service.go

```diff
@@ -141,6 +143,7 @@ type service struct {
 	lggr                logger.Logger
 	version             string
 	loopRegistrarConfig plugins.RegistrarConfig
+	syncNodeInfoCancel  context.CancelFunc
```

Comment:
I think instead of using this to pass context, we should just have `syncNodeInfoWithRetry` accept a context as parameter, each of the caller should have a context value to pass it in.